Updated Avatar Style and `Hide Chat Avatars` title text.

DeclineThyself opened the pull request:
I found the `Hide Chat Avatars` toggle only after creating the previous version of this commit.
I rewrote the commit because I believe this is better UX.

This PR removes the toggle and migrates user settings. 

## Checklist:

- [X] I have read the [Contribution guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

Commit:
Added an option to hide message avatars in `User Settings > UI Theme`.

Commit:
Deprecated the `Hide Chat Avatars` toggle.
Moved the option to hide message avatars to `User Settings > UI Theme > Avatars`.

DeclineThyself commented:
/gemini review

DeclineThyself commented on public/scripts/power-user.js:
```diff
@@ -493,9 +493,18 @@ function switchMesIDDisplay() {
+        saveSettingsDebounced();
+    }
+    //This doesn't need to be immediately saved.
+    power_user.hideChatAvatars_enabled = undefined;
```
An else block would be more lines.

DeclineThyself commented:
/gemini summary

Wolfsblvt reviewed (commented):
Not sure if this is a good decision.
Avatars will still be used and display, even if you don't want them in chat.
Same avatar style gets used for the favorites bar, in the char list, to display in the persona connections, etc etc.

What would the style chosen be if you switch the selector to "hidden"?
How could you hide in chat but still decide to use circles in all other places?

Honestly, if someone would request the feature and it weren't already an option, I'd say "hideChatAvatars" is a simple Custom CSS tweak, and does not need to be a toggle.

DeclineThyself commented:
> Not sure if this is a good decision.

Thanks for explaining, I didn't consider those use-cases.
What do you think of my previous version? This leaves `Hide Chat Avatars` intact and adds an additional option to the drop down.
https://github.com/SillyTavern/SillyTavern/pull/4908/changes/f6b8aedb54b6e0372d3d41f75b013bbe86a814bc

This should allow for more flexibility instead of less, but I'm likely missing something.

Cohee1207 commented:
What is the intent behind a change? You want to hide _all_ avatars globally? Since "Hide Chat Avatars" is explicitly about chat avatars, and the dropdown is a global avatar style (including lists, quick swaps, etc), I don't think it makes sense to be included into the dropdown.

Wolfsblvt commented:
I think, tbh, the "Hide chat avatars" toggle should just be removed, because in what we usually do in the last months or even years, this option would not warrant a separate UI toggle.
Though considering it's already there will lead this to something more like... leave it as is, for historic reasons..

Cohee1207 commented:
I use it sometimes, it's useful for making more compact screenshots. I'm not entirely against it being a separate option.

Commit:
Revert "Deprecated the `Hide Chat Avatars` toggle."

This reverts commit 6f616767cce4aa2c27a3a53543687d6eaf379758.

Commit:
Moved `Hide Chat Avatars` next to `Avatar Style`, added title text.

DeclineThyself commented:
> I use it sometimes, it's useful for making more compact screenshots. I'm not entirely against it being a separate option.

[I reverted the deprecation.](https://github.com/SillyTavern/SillyTavern/pull/4908/commits/34d53ed260d9110e6a79480c161877bd5f5bcab5)

> Since "Hide Chat Avatars" is explicitly about chat avatars, and the dropdown is a global avatar style (including lists, quick swaps, etc).

I didn't realize that. I've added a title that clarifies the difference and moved the toggle.
[How's this?](https://github.com/SillyTavern/SillyTavern/pull/4908/changes/a64f045cb1982c282db9bb79250c020eabf6e441#diff-c2cc24bc9001b11b6add48a4cd8f893d5d6c6e4d1bd254158bd14ab997f552cdR4770-R4783)
```html
      <div class="flex-container alignItemsBaseline" title="This style applies to all avatars globaly, including your Persona, Character Managment, Account selection, etc." data-i18n="[title]This style applies to all avatars globaly, including your Persona, Character Managment, Account selection, etc.">
```

> You want to hide all avatars globally?

Yes, I'd like to add that as an option.

>  because in what we usually do in the last months or even years, this option would not warrant a separate UI toggle.

Would it warrant another option in the drop-down?

How are these [changes](https://github.com/SillyTavern/SillyTavern/pull/4908/changes)? Should any be removed or altered?

Cohee1207 commented on public/index.html:
```diff
@@ -4767,15 +4767,20 @@ <h4 class="title_restorable">
+                                        <option value="4" data-i18n="Hidden">Hidden</option>
                                     </select>
                                 </div>
+                                <label for="hideChatAvatarsEnabled" class="checkbox_label" title="Hide avatars, only in chat messages. data-i18n="[title]Hide avatars, only in chat messages.>
```
I don't want to mix checkboxes and selects in one section. Move it back please

DeclineThyself replied:
Fixed: https://github.com/SillyTavern/SillyTavern/pull/4908/commits/eccc346a78fb321d4190c73b48316754220302cd

Cohee1207 commented:
This will also remove avatars from the following sections, making some parts of functionality totally inoperable:

1. Hotswaps (there's a dedicated toggle for it)
2. Persona connections list
3. Character editor (making it impossible to edit character image)
4. "Tags as folders" character avatar lists
5. Tag import avatar lists

And probably something else, but that's what I could think of right away.

Commit:
Reverted changes.

DeclineThyself commented:
> This will also remove avatars from the following sections, making some parts of functionality totally inoperable:
> 
>     1. Hotswaps (there's a dedicated toggle for it)
> 
>     2. Persona connections list
> 
>     3. Character editor (making it impossible to edit character image)
> 
>     4. "Tags as folders" character avatar lists
> 
>     5. Tag import avatar lists
> 
> 
> And probably something else, but that's what I could think of right away.

@Cohee1207, Thank you, I missed all of those.
I didn't know about those features.

This PR now only updates the hover titles, which may help prevent others from being confused like I was.

Are these changes okay?
https://github.com/SillyTavern/SillyTavern/pull/4908/changes#diff-c2cc24bc9001b11b6add48a4cd8f893d5d6c6e4d1bd254158bd14ab997f552cdR4770

Cohee1207 commented:
Yeah, that's fine. Perhaps a documentation update on User Settings would also be useful.

Cohee1207 reviewed (approved):
Looks good to me.

cloak1505 commented on public/index.html:
```diff
@@ -4954,7 +4943,7 @@ <h4 class="title_restorable">
                                     <small data-i18n="Message IDs">Message IDs</small>
                                 </label>
-                                <label for="hideChatAvatarsEnabled" class="checkbox_label" title="Hide avatars in chat messages." data-i18n="[title]Hide avatars in chat messages.">
+                                <label for="hideChatAvatarsEnabled" class="checkbox_label" title="Hide avatars, only in chat messages. data-i18n="[title]Hide avatars, only in chat messages.>
```
`data-i18n=` shows up in hover text because double quotes are missing after `only in chat messages.`

Cohee1207 replied:
Fixed. 
https://github.com/SillyTavern/SillyTavern/commit/e7c9496a619d600ce8bf2548b80d9d8561e16d74

DeclineThyself replied:
Thank you.
